In [17]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import warnings



In [23]:
import numpy as np
from math import factorial

##1.Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. 
Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.Даны значения зарплат из выборки выпускников:

In [2]:
salary = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70, 75, 65, 84, 90, 150]

In [3]:
AA=0
for item in salary:
    AA = AA + item
MO = AA/len(salary)
print("Математическое ожидание:", MO)

Математическое ожидание: 73.14285714285714


Найдём среднее квадратичное отклонение

In [4]:
AA = 0
for item in salary:
    AA = AA + ((item - MO)**2)
QO = (AA/len(salary))**0.5
print("Квадратичное отклонение:", QO)

Квадратичное отклонение: 46.20686994302874


Найдём смещённую и несмещённую оценки дисперсии

In [5]:
AA = 0
for item in salary:
    AA = AA + ((item - MO)**2)
D_biased = AA/len(salary)
D_unbiased = AA/(len(salary)-1)
print("Смещенная дисперсия:", D_biased)
print("Несмещенная дисперсия:", D_unbiased)

Смещенная дисперсия: 2135.074829931973
Несмещенная дисперсия: 2241.8285714285716


##2.В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

Пусть A-искомое событие "из двух корзин вытащили 3 белых" мяча. Тогда это событие может произойти следующими способами:
а) B - из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча;
б) C - из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча;
в) D - из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч.
Вероятность события A будет определяться как сумма вероятностей этих событий.

Определим вероятность события B

In [11]:
def combinations(n,k):
    return factorial(n)/(factorial(k)*factorial(n-k))

In [26]:
P1=(combinations(3,2)/combinations(8,2))*(combinations(5,3)*combinations(7,1)/combinations(12,4))
print(f'Вероятность события "из первой корзины не вытащили ни одного белого мяча'
      f', из второй вытащили 3 белых мяча"\nP(B) = {P1: .4f}'
     )

Вероятность события "из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча"
P(B) =  0.0152


Определим вероятность события C 

In [27]:
P2=(combinations(5,1)*combinations(3,1)*combinations(5,2)*combinations(7,2))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 1 белый мяч'
      f', из второй вытащили 2 белых мяча"\nP(C) = {P2: .4f}'
     )

Вероятность события "из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча"
P(C) =  0.2273


Определим вероятность события D

In [28]:
P3=(combinations(5,2)*combinations(5,1)*combinations(7,3))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 2 белых мяча'
      f', из второй вытащили 1 белый мяч"\nP(D) = {P3: .4f}'
     )


Вероятность события "из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч"
P(D) =  0.1263


Вычислим событие А

In [29]:
print(f'Вероятность того, что из вытащенных мячей 3 белые Р(А) = {P1+P2+P3: .4f}')

Вероятность того, что из вытащенных мячей 3 белые Р(А) =  0.3687


Ответ: вероятность того, что 3 мяча белые = 36,87%

##3.На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

Рассмотрим следующие события:
A-в результате выстрела, цель поражена;
B1 выстрел произведен первым спортсменом;
B2 выстрел произведен вторым спортсменом;
B3 выстрел произведен третьим спортсменом.
Вероятности того, что выстрел произведен определенным спортсменом равны, т.к. выстрел 1, и произвести его могут только 3 этих спортсмена, т.е. события B1,B2,B3 составляют полную группу событий.

In [30]:
PB=1/3
PA=PB*0.9+PB*0.8+PB*0.6
print(f'Полная вероятность наступления события А Р(А) = {PA: .4f}')

Полная вероятность наступления события А Р(А) =  0.7667


исходя из того, что событие А уже произошло, с помощью формулы Байеса найдём вероятность выстрелов спортсменов

In [31]:
PAB1=PB*0.9/PA
PAB2=PB*0.8/PA
PAB3=PB*0.6/PA
print(f'Вероятность того, что выстрел произвёл первый спортсмен: {PAB1: .4f};\n'
      f'Вероятность того, что выстрел произвёл второй спортсмен: {PAB2: .4f};\n'
      f'Вероятность того, что выстрел произвёл третий спортсмен: {PAB3: .4f}.'
     )

Вероятность того, что выстрел произвёл первый спортсмен:  0.3913;
Вероятность того, что выстрел произвёл второй спортсмен:  0.3478;
Вероятность того, что выстрел произвёл третий спортсмен:  0.2609.


Ответ: вероятность того, что выстрел произведен: a)первым спортсменом = 39,13% б) вторым спортсменом = 34,78% в) третьим спортсменом = 26,09%.

##4.В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

Пусть X - количество студентов на факультете A, тогда на факультете B учится X студентов, а на факультете C 2X студентов.
Введем события SA,SB,SC, что сессию сдаёт студент факультетов A,B,C соответственно, и событие D-сессия сдана.

In [32]:
PD=0.25*0.8+0.25*0.7+0.5*0.9
print(f'Полная вероятность наступления события D, P(D) = {PD}.')

Полная вероятность наступления события D, P(D) = 0.825.


По формуле Байеса вычислим вероятность того, на каком факультете учится студент, успешно сдавший сессию.

In [33]:
PDSA=0.25*0.8/PD
PDSB=0.25*0.7/PD
PDSC=0.5*0.9/PD
print(f'Вероятность того, что студент учится на факультете А: {PDSA: .4f};\n'
      f'Вероятность того, что студент учится на факультете B: {PDSB: .4f};\n'
      f'Вероятность того, что студент учится на факультете C: {PDSC: .4f}.'
      )

Вероятность того, что студент учится на факультете А:  0.2424;
Вероятность того, что студент учится на факультете B:  0.2121;
Вероятность того, что студент учится на факультете C:  0.5455.


Ответ: вероятность, что студент учится: a) на факультете A = 24,24% б) на факультете B = 21,21% в) на факультете C =54,55%

##5.Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?


In [34]:
P3=0.1*0.2*0.25
print(f'Вероятность того, что из строя выйдут все детали Р(3) = {P3: .4f}')

Вероятность того, что из строя выйдут все детали Р(3) =  0.0050


In [35]:
P2=0.1*0.2*0.75+0.1*0.25*0.8+0.2*0.25*0.9
print(f'Вероятность того, что из строя выйдут только 2 детали Р(2) = {P2: .4f}')

Вероятность того, что из строя выйдут только 2 детали Р(2) =  0.0800


In [36]:
P0=0.9*0.8*0.75
print(f'Вероятность того, что из строя не выйдет ни одной детали Р(0) = {P0: .4f}')

Вероятность того, что из строя не выйдет ни одной детали Р(0) =  0.5400


In [37]:
P_0=1-P0
print(f'Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) = {P_0: .4f}')

Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) =  0.4600


In [38]:
P1=0.1*0.8*0.75+0.2*0.9*0.75+0.25*0.9*0.8
print(f'Вероятность того, что выйдет из строя одна деталь Р(1) = {P1: .4f}')

Вероятность того, что выйдет из строя одна деталь Р(1) =  0.3750


In [39]:
P=0.3750 + 0.08
print(f'Вероятность того, что из строя выйдут от одной до двух деталей Р = {P: .4f}')

Вероятность того, что из строя выйдут от одной до двух деталей Р =  0.4550


Ответ: вероятность того, что в первый месяц выйдут из строя: а)все детали = 0,5% б)только две детали = 8% в) хотя бы одна деталь = 46% г)от одной до двух деталей = 45,5%